### PR TITLE
Restrict `.providerSpecific.location` value to enum

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/roadmap/issues/2573

To avoid bad values (especially unsupported regions), this PR restricts the location value to a set of defined values, taken from https://github.com/giantswarm/capi-image-builder/blob/v1.6.10/helm/capi-image-builder/values.yaml#L48-L52

This also means that when the list in `capi-image-builder` gets changed, it should be changed here accordingly.

### Please check if PR meets these requirements

- [x] Results of the diffs have been examined and no unintended changes are being introduced.

### Helper

* to disable the GH Action generating manifests diffs for installations, between target and source branches, comment the `/no_diffs_printing` on the PR.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: helm/cluster-azure/README.md [hunk 1]
```diff
@@ -102,6 +102,8 @@ Properties within the `.metadata` top-level object
 | **Property** | **Description** | **More Details** |
 | :----------- | :-------------- | :--------------- |
 | `metadata.description` | **Cluster description** - User-friendly description of the cluster's purpose.|**Type:** `string`<br/>|
+| `metadata.labels` | **Labels** - These labels are added to the Kubernetes resourses defining this cluster.|**Type:** `object`<br/>|
```
Comment (fiunchinho) [inline helm/cluster-azure/README.md:105]: I believe this is unrelated to the PR.

File: helm/cluster-azure/values.schema.json [hunk 2]
```diff
@@ -514,6 +514,14 @@
                 "location": {
                     "type": "string",
                     "title": "Location",
+                    "$comment": "Valid options from https://github.com/giantswarm/capi-image-builder/blob/v1.6.10/helm/capi-image-builder/values.yaml#L48-L52",
```
Comment (fiunchinho) [inline helm/cluster-azure/values.schema.json:517]: Can you maybe add a comment in that `capi-image-builder` file regarding updating `cluster-azure` when adding a new region?

Comment (marians) [reply helm/cluster-azure/values.schema.json:517]: All right, added a draft PR here. https://github.com/giantswarm/capi-image-builder/pull/91

File: helm/cluster-azure/README.md (hunk 1 above)
Comment (marians) [reply helm/cluster-azure/README.md:105]: True. I just updated the docs automatically, and this was added, too.

**Review by fiunchinho — APPROVED** [review]

**Comment by fiunchinho** [comment]
`heimdall` waits until this check is green `/run cluster-test-suites`, but every time a new commit is added, those checks are gone.
